File: packages/sdk-js/examples/webhook.ts
/**
 * @smscode/sdk — verify an incoming webhook (framework-agnostic).
 * Pass the EXACT received body bytes (never re-serialized), the
 * `X-Webhook-Signature` header, and your configured webhook secret.
 * In your app, import from "@smscode/sdk"; this in-repo example imports the
 * local source so it type-checks in CI.
 */
import { verifyWebhookSignature, parseWebhookEvent } from "../src/index.js";

export async function handleWebhook(rawBody: string, signatureHeader: string | null, secret: string) {
  // MUST await — verifyWebhookSignature returns Promise<boolean>. Without await a
  // (truthy) Promise would accept EVERY request, including forged ones.
  const ok = await verifyWebhookSignature(rawBody, signatureHeader ?? "", secret);
  if (!ok) return { status: 401 as const }; // reject: bad/missing signature
  const event = parseWebhookEvent(rawBody);
  console.log("verified webhook event:", event.event);
  return { status: 200 as const };
}

File: packages/sdk-py/examples/webhook.py
from smscode import parse_webhook_event, verify_webhook_signature


def handle_webhook(raw_body: bytes, signature_header: str | None, secret: str) -> tuple[int, str]:
    if not verify_webhook_signature(raw_body, signature_header or "", secret):
        return 401, "bad signature"

    event = parse_webhook_event(raw_body)
    if event["event"] == "order.otp_received":
        otp_code = event["data"].get("otp_code")
        otp_message = event["data"].get("otp_message")
        if otp_code:
            print("OTP:", otp_code)
        else:
            print("SMS received without a classified code:", otp_message)
    return 204, "ok"
